multiversx/mx-chain-go
# Trie sync progress status

## Reasoning behind the pull request
- Add trie sync progress status percentage
  
## Proposed changes
- Added a new check into gateway in order to fetch estimated number of trie nodes from last snapshot
- export gateway url from config on bootstrap metrics
- Added additional config section in `config.toml` to specify gateway url

## Testing procedure
- Standard system test
- Check in termui that trie sync progress percentage works, on testnet, devnet or mainnet (it can be tested also on internal testnet if that's feasable)
     - set gateway URL from newly added section in `config.toml`
          - start termui while node is syncing and check trie sync area
     - if gateway URL is not set, it should not crash, it should show N/A on trie sync progress section

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
